File: fay-frontend/src/config/openaiTtsVoices.ts
import type { VoiceOption } from '../types';

export interface NormalizedVoiceOption {
  value: string;
  label: string;
}

export const OPENAI_CHINESE_TTS_VOICES: NormalizedVoiceOption[] = [
  { value: 'zh-CN-XiaoxiaoNeural', label: '晓晓（女声）' },
  { value: 'zh-CN-XiaoyiNeural', label: '晓伊（女声）' },
  { value: 'zh-CN-YunjianNeural', label: '云健（男声）' },
  { value: 'zh-CN-YunxiNeural', label: '云希（男声）' },
  { value: 'zh-CN-YunxiaNeural', label: '云夏（男声）' },
  { value: 'zh-CN-YunyangNeural', label: '云阳（男声）' },
  { value: 'zh-CN-liaoning-XiaobeiNeural', label: '小贝（辽宁女声）' },
  { value: 'zh-CN-shaanxi-XiaoniNeural', label: '晓妮（陕西女声）' },
  { value: 'zh-HK-HiuGaaiNeural', label: '晓佳（香港女声）' },
  { value: 'zh-HK-HiuMaanNeural', label: '晓曼（香港女声）' },
  { value: 'zh-HK-WanLungNeural', label: '云龙（香港男声）' },
  { value: 'zh-TW-HsiaoChenNeural', label: '晓臻（台湾女声）' },
  { value: 'zh-TW-HsiaoYuNeural', label: '晓雨（台湾女声）' },
  { value: 'zh-TW-YunJheNeural', label: '云哲（台湾男声）' },
];

function normalizeVoice(voice: VoiceOption): NormalizedVoiceOption {
  const value = voice.value || voice.id || '';
  return {
    value,
    label: voice.label || voice.name || value,
  };
}

function isChineseNeuralVoice(value: string) {
  return /^zh-[A-Za-z-]+Neural$/.test(value);
}

function isChineseNeuralVoiceList(voices: NormalizedVoiceOption[]) {
  return voices.length > 0 && voices.every((voice) => isChineseNeuralVoice(voice.value));
}

export function normalizeVoiceOptions(voices: VoiceOption[] = []): NormalizedVoiceOption[] {
  const normalized = voices.map(normalizeVoice).filter((voice) => voice.value);
  if (isChineseNeuralVoiceList(normalized)) {
    return OPENAI_CHINESE_TTS_VOICES.map((voice) => ({ ...voice }));
  }
  return normalized;
}

File: tts/openai_tts_voices.py
OPENAI_TTS_VOICE_LIST = [
    {"id": "zh-CN-XiaoxiaoNeural", "name": "晓晓（女声）"},
    {"id": "zh-CN-XiaoyiNeural", "name": "晓伊（女声）"},
    {"id": "zh-CN-YunjianNeural", "name": "云健（男声）"},
    {"id": "zh-CN-YunxiNeural", "name": "云希（男声）"},
    {"id": "zh-CN-YunxiaNeural", "name": "云夏（男声）"},
    {"id": "zh-CN-YunyangNeural", "name": "云阳（男声）"},
    {"id": "zh-CN-liaoning-XiaobeiNeural", "name": "小贝（辽宁女声）"},
    {"id": "zh-CN-shaanxi-XiaoniNeural", "name": "晓妮（陕西女声）"},
    {"id": "zh-HK-HiuGaaiNeural", "name": "晓佳（香港女声）"},
    {"id": "zh-HK-HiuMaanNeural", "name": "晓曼（香港女声）"},
    {"id": "zh-HK-WanLungNeural", "name": "云龙（香港男声）"},
    {"id": "zh-TW-HsiaoChenNeural", "name": "晓臻（台湾女声）"},
    {"id": "zh-TW-HsiaoYuNeural", "name": "晓雨（台湾女声）"},
    {"id": "zh-TW-YunJheNeural", "name": "云哲（台湾男声）"},
]


def get_openai_tts_voice_list():
    return [voice.copy() for voice in OPENAI_TTS_VOICE_LIST]
